### PR TITLE
Return error upon attemping to create named objects in PAL

### PR DESCRIPTION
**Pull request by kouvel**
Update PAL APIs that create named objects (mutex, semaphore, event, file mapping) to return a not-supported error code. It was decided to not support cross-process synchronization in PAL at present time due to complexities involved in reliably emulating Windows' behavior. @stephentoub has already made changes on the FX side to throw PlatformNotSupportedException in these cases.

Related to issue #1237.

### CODE REVIEW

**Comment by kouvel** [comment]
CC @sergiy-k, @jkotas

File: src/md/enc/stgio.cpp [hunk 1]
```diff
@@ -869,7 +869,7 @@ HRESULT StgIO::MapFileToMem(            // Return code.
 #endif
             
             if ((m_hMapping = WszCreateFileMapping(m_hFile, pAttributes, dwProtectionFlags,
-                0, 0, *m_rcShared ? m_rcShared : 0)) == 0)
+                0, 0, nullptr)) == 0)
```
Comment (jkotas) [inline src/md/enc/stgio.cpp]: It looks like m_rcShared is never assigned to non-empty string - could you please delete it from everywhere?

**Comment by jkotas** [comment]
LGTM otherwise.

File: src/md/enc/stgio.cpp (hunk 1 above)
Comment (kouvel) [inline src/md/enc/stgio.cpp]: Done, and squashed commits

File: src/md/enc/stgio.cpp [hunk 2]
```diff
@@ -790,7 +789,7 @@ HRESULT StgIO::MapFileToMem(            // Return code.
     // Note that any shared memory objects will require the handles to be in place
```
Comment (jkotas) [inline src/md/enc/stgio.cpp]: Nit - this comment is no longer relevant

Comment (kouvel) [inline src/md/enc/stgio.cpp]: Ah yes, removed

**Comment by sergiy-k** [comment]
LGTM

**Pull request merged**